controllers/cassandrabackup_controller.go

```diff
@@ -153,8 +204,44 @@ func (r *CassandraBackupReconciler) Reconcile(req ctrl.Request) (ctrl.Result, er
 			}
 		}(p)
 	}
+	return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+}
 
-	return ctrl.Result{}, nil
+func (r *CassandraBackupReconciler) removeBackup(ctx context.Context, backup *api.CassandraBackup) error {
+	cassdcKey := types.NamespacedName{Namespace: backup.Namespace, Name: backup.Spec.CassandraDatacenter}
+	cassdc := &cassdcapi.CassandraDatacenter{}
+	err := r.Get(ctx, cassdcKey, cassdc)
+	if err != nil {
+		return err
+	}
+
+	pods, err := r.getCassandraDatacenterPods(ctx, cassdc)
+	if err != nil {
+		return err
+	}
+
+	success := false
+	// Try all pods, a single success is enough
+	for _, pod := range pods {
+		if !hasMedusaSidecar(&pod) {
+			// Lets try next one, we only need to call a single pod to delete the backups
+			continue
+		}
+		if err := deleteBackup(ctx, backup.Spec.Name, &pod, r.ClientFactory); err == nil {
+			success = true
```

Comment:
@burmanm in my previous comment I mentioned that it seemed like `removeBackup` was getting called for each pod.   It is. There should a `break` after line 231.